src/logic/prepare/provider/provider.ts

```diff
@@ -0,0 +1,194 @@
+import type {
+  ProviderEntry,
+  SecurityValues,
+  SuperJsonDocument,
+} from '@superfaceai/ast';
+import {
+  prepareProviderParameters,
+  prepareSecurityValues,
+} from '@superfaceai/ast';
+import { mergeProvider, NodeFileSystem } from '@superfaceai/one-sdk';
+import { ServiceApiError } from '@superfaceai/service-client';
+import inquirer from 'inquirer';
+
+import type { ILogger } from '../../../common';
+import { UNVERIFIED_PROVIDER_PREFIX } from '../../../common';
+import type { UserError } from '../../../common/error';
+import { fetchProviderInfo } from '../../../common/http';
+import { OutputStream } from '../../../common/output-stream';
+import { resolveSuperfaceRelativePath } from '../../../common/path';
+import * as providerTemplate from '../../../templates/provider';
+import { selecetBaseUrl } from './base-url';
+import { selectIntegrationParameters } from './parameters';
+import { selectSecurity } from './security';
+
+export async function prepareProvider(
+  {
+    provider,
+    superJson,
+    superJsonPath,
+    options,
+  }: {
+    provider: string;
+    superJson: SuperJsonDocument;
+    superJsonPath: string;
+    options?: {
+      force?: boolean;
+      station?: boolean;
+    };
+  },
+  { logger, userError }: { logger: ILogger; userError: UserError }
+): Promise<void> {
+  if (
+    superJson.providers?.[provider] !== undefined &&
+    options?.force !== true
+  ) {
+    logger.warn('providerAlreadyExists', provider);
+
+    return;
+  }
+  // TODO: if provider is local and without unverified prefix - rename?
+
+  // check remote
+  let name: string = provider;
+  if (!provider.startsWith(UNVERIFIED_PROVIDER_PREFIX)) {
+    try {
+      const remote = await fetchProviderInfo(name);
+
+      const useRemote = (
+        await inquirer.prompt<{ continue: boolean }>({
+          name: 'continue',
+          message: `Provider "${provider}" found in Superface registry:\n${JSON.stringify(
+            remote,
+            undefined,
+            2
+          )}\nDo you want to use it?`,
+          type: 'confirm',
+          default: true,
+        })
+      ).continue;
+
+      if (useRemote) {
+        // set up values and add provider to super json
+        await updateSuperJson(
+          {
+            name,
+            superJson,
+            superJsonPath,
+            security: prepareSecurityValues(name, remote.securitySchemes ?? []),
+            parameters: prepareProviderParameters(
+              name,
+              remote.parameters ?? []
+            ),
+          },
+          { logger }
+        );
+
+        return;
+      }
+    } catch (error) {
```

Comment:
This try-catch mechanism of discovering potential provider in registry would look nicer if abstracted in a function. Then the logic here would look cleaner.